src/components/DropdownMenu.stories.tsx

```diff
@@ -0,0 +1,105 @@
+import { MouseEvent } from 'react';
+import { DropdownMenu, DropdownMenuItemButton, useDropdownMenu } from './DropdownMenu';
+
+const onClick = (e: MouseEvent<HTMLButtonElement>) => {
+  console.log(`Activated "${e.currentTarget.innerText}"`);
+};
+
+export const Primary = () => {
+  const enabledState = useDropdownMenu();
+  const disabledState = useDropdownMenu({ disabled: true });
+
+  return <>
+    <DropdownMenu id='enabled-grades-menu' state={enabledState} text='Sync/Export Grades' variant='primary'>
+      <DropdownMenuItemButton onClick={onClick} state={enabledState}>
```

Comment:
If the MenuItemButtons should always have the same state as their enclosing DropdownMenus, you should consider making that a Context.

Reply:
I'll give that a try

Reply:
if you do as roy suggests you can also move the `useDropdownMenu` call and state provider into the dropdown menu component, removing the burden of managing that from the consumer call

Reply:
Added the context provider and removed export for useDropdownMenu